iterm2_dwim/__init__.py

```diff
@@ -5,12 +5,13 @@
 import sys
 import traceback
 
-from iterm2_dwim.editors import emacs
+from iterm2_dwim.editors import emacs, sublime
+from iterm2_dwim.settings import editor
 from iterm2_dwim.logger import log
 from iterm2_dwim.parsers import get_path_and_line
 
 
-Editor = emacs.Editor
+Editor = sublime.Sublime if editor.lower() == 'sublime' else emacs.Emacs
```

Comment:
I put this in `main`.

I also used your example to instantiate the Editor classes with `settings.<path to client>`, and made `path` and `line` arguments to each editor's `visit_file` method.